### ISSUE TITLE
Mulitple tabs with a different database connection active in each

### ISSUE THREAD

**Issue by oraclerob** [HeidiSQL/HeidiSQL#1559]
**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**
For each tab that is open -  a database connection session is active (with the database previously selected "in use" in the tree) and tied to that tab unless you click on the database tree and change it.  This is similar behavior to how TOAD works and allows the user to quickly query from multiple tabs with different connections without needing to reselect the new database from the tree - which is annoying and also not intuitive as the current database selected is not displayed in the tab description (which TOAD does as well).

**Describe alternatives you've considered**
The startup reconnect database preference DOES NOT solve this.

**Additional context**
Adding this feature will allow multiple queries to be run at once in the future. I do realise that this might involve some threading and display UI considerations, but I think this would speed up productivity immensely when one is working with several connections at once which is more than likely on a larger system.

**Comment by ansgarbecker** [comment]
That was requested before, while I can't find the issue id right now.
Issue #932 is related, at least to get a hint to which connection a query belongs.
Did you already use the session background color?

![grafik](https://user-images.githubusercontent.com/7986591/163536740-a948d404-bc60-4832-9c5a-3e9e95f7d43a.png)

**Comment by oraclerob** [comment]
Thanks for your response. No I don't think you understand what I'm asking for, your screenshot shows 1 tab open - I need many open with different active connections. Simply to have multiple SQL query tabs open each with their own connection with database selected. DBeaver 22 has this functionality and unfortunately I've had to move over to using this product as it is impossible to work with many connections and tabs open and have to remember which database is connected to which tab, then have to select the database you are working on whenever you switch tabs. It's a pity, as I like HeidiSQL but it needs to be able to have many connections open at once, with the tab connected to that connection (always) - to be usable in larger enterprises otherwise big mistakes can occur if the current tab is connected to the wrong database. I would be willing to move back to this product and even donate if this functionality could be added.

**Comment by ansgarbecker** [comment]
Well I understood it quite well. The color in the editor gutter is just to to avoid to "_accidentally run a statement not intended for that database if you have the wrong connection selected in the tree_".

Binding tabs to connection requires some UI changes, without taking much space. Do you have an idea how that could look like on the tabs, or wherever?

**Comment by oraclerob** [comment]
Yes I sure do have an idea. Something like this <servername>-<database-connection>  instead of "Query #" in the name of the tab initially. The connection is then opened from the tree (as a selection in the right click) and the connection and database is forever tied to that query tab (unless you specifically change it - but I wouldn't even bother with that for now because nobody will ever really do that). When you save or exit, this is remembered on startup.  This is similar to how DBeaver (and TOAD) works and so far for me it has increased productivity and reduced accidental mistakes immensely.

![image](https://user-images.githubusercontent.com/11176397/163543101-e14b06e7-0219-4124-8326-b562ea2223e1.png)

**Comment by ansgarbecker** [comment]
Yes, looks like a good idea. I need a better tab component, which probably supports a kind of pulldown within the tabs to be able to switch to another connection. Colored tabs and native close buttons would be a nice benefit too, as these are currently implemented in a dirty way.
